### ISSUE TITLE
vmsingle/vmselect panic for negative lookbehind window

### ISSUE THREAD

**Issue by Amper** [VictoriaMetrics/VictoriaMetrics#4795]
### Describe the bug

vmsingle/vmselect panics for the query:

```
max_over_time(time()[10m-12m])
```

### To Reproduce

To reproduce, launch the single-node instance and issue a query with a negative window:

```
max_over_time(time()[10m-12m])
```

### Version

v 1.92.1

### Logs

```
2023-08-07T17:44:05.141Z	panic	VictoriaMetrics/app/vmselect/promql/rollup.go:640	BUG: Window must be non-negative; got -120000
panic: BUG: Window must be non-negative; got -120000

goroutine 452 [running]:
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logMessage({0xbd6a4a, 0x5}, {0xc0000326c0, 0x2d}, 0x1?)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:281 +0xa96
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevelSkipframes(0x1, {0xbd6a4a, 0x5}, {0xbf732d?, 0x0?}, {0xc00120fbe0?, 0xc0003f4000?, 0x67bd34?})
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:138 +0x1a5
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.logLevel(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:130
github.com/VictoriaMetrics/VictoriaMetrics/lib/logger.Panicf(...)
	github.com/VictoriaMetrics/VictoriaMetrics/lib/logger/logger.go:126
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.(*rollupConfig).doInternal(0xc00021c480, {0x0, 0x0, 0x0}, 0x0, {0xc00018e800, 0x4, 0x4}, {0xc00018e820, 0x4, ...})
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/rollup.go:640 +0x254
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.(*rollupConfig).Do(...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/rollup.go:619
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.doRollupForTimeseries({0xc00018e6e0, 0xd}, 0x0, 0xc00021c480, 0xc00021e0e0, 0x0?, {0xc00018e800, 0x4, 0x4}, {0xc00018e820, ...}, ...)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1279 +0x4a8
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.evalRollupFuncWithSubquery.func1(0xc00021e000, {0x0?, 0x0?, 0x0?}, {0x0?, 0xbe8a78?, 0x0?}, 0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:943 +0x575
github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.doParallel.func1(0x0)
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:1004 +0x104
created by github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql.doParallel
	github.com/VictoriaMetrics/VictoriaMetrics/app/vmselect/promql/eval.go:999 +0x1ab
```

**Comment by valyala** [comment]
The panic should be fixed in the commit 072d891ed9db460473c1bdc23300cd15d837952d . This commit will be included in the next release.

**Comment by valyala** [comment]
The panic for negative lookbehind window in [rollup functions](https://docs.victoriametrics.com/MetricsQL.html#rollup-functions) has been fixed starting from the release [v1.93.0](https://docs.victoriametrics.com/CHANGELOG.html). Closing the issue then.
